src/core/gui.py

```diff
@@ -219,6 +232,12 @@ def run(self):
 
 # Test the GUI with random numbers generation
 if __name__ == '__main__':
+
+    # Read YAML file 
+    config_file_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), 'config\config.yml')
```

Comment:
os.path.join should take care of adding all the slashes to allow use on different os. 
this syntax may be useful: os.path.abspath(os.path.join(os.path.dirname(__name__), 'config', 'config.yml'))

Reply:
see above